src/pytorch_lightning/core/datamodule.py

```diff
@@ -53,7 +60,7 @@ def teardown(self):
                 # called on every process in DDP
     """
 
-    name: str = ...
+    name = ...
```

Comment:
```suggestion
    name: str
```
We still want to have it typed, but the assignment is not necessary

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13693/commits/2f6e162ab7e399e38b61d4ff57e2982fe29eceb7